scripts/mvftoms.py

```diff
@@ -241,47 +252,57 @@ def _cp_index(a1, a2, pol):
     # otherwise finally select any of HH,VV present (the default).
     pols_to_use = ['HH', 'HV', 'VH', 'VV'] if (options.full_pol or options.circular) else \
         list(np.unique(options.pols_to_use.split(','))) if options.pols_to_use else \
-        [pol for pol in ['HH', 'VV'] if pol in pols_in_file]
+        [pol for pol in ['HH', 'VV'] if pol in pols_in_dataset]
 
     # Check we have the chosen polarisations
-    if np.any([pol not in pols_in_file for pol in pols_to_use]):
+    if np.any([pol not in pols_in_dataset for pol in pols_to_use]):
         raise RuntimeError("Selected polarisation(s): %s not available. "
                            "Available polarisation(s): %s"
-                           % (','.join(pols_to_use), ','.join(pols_in_file)))
+                           % (','.join(pols_to_use), ','.join(pols_in_dataset)))
 
     # Set full_pol if this is selected via options.pols_to_use
     if set(pols_to_use) == set(['HH', 'HV', 'VH', 'VV']) and not options.circular:
         options.full_pol = True
 
-    pol_for_name = 'full_pol' if options.full_pol else \
-                   'circular_pol' if options.circular else \
-                   '_'.join(pols_to_use).lower()
-
+    # Extract one MS per spectral window in the dataset(s)
     for win in range(len(dataset.spectral_windows)):
         dataset.select(reset='T')
 
-        # Extract MS file per spectral window in observation file
         freq_MHz = dataset.spectral_windows[win].centre_freq / 1e6
         print('Extract MS for spw %d: central frequency %.2f MHz' % (win, freq_MHz))
 
-        # If no output MS filename supplied, infer the output filename
-        # from the first dataset.
+        # If no output MS directory name supplied, infer it from first dataset
         if options.output_ms is None:
-            # create MS in current working directory
-            ms_name = '%s_%d.%s%s.ms' % (
-                os.path.splitext(args[0])[0], freq_MHz,
-                "" if len(args) == 1 else "et_al.", pol_for_name)
+            url_parts = urllib.parse.urlparse(args[0], scheme='file')
+            # Create MS in current working directory (strip off directories)
+            dataset_filename = os.path.basename(url_parts.path)
+            # Get rid of the ".full" bit on RDB files (it's the same dataset)
+            if dataset_filename.endswith('.full.rdb'):
+                dataset_basename = dataset_filename[:-9]
+            else:
+                dataset_basename = os.path.splitext(dataset_filename)[0]
+            # Add frequency to name to disambiguate multiple spectral windows
+            if len(dataset.spectral_windows) > 1:
+                dataset_basename += '_{:d}MHz'.format(int(freq_MHz))
+            # Add ".et_al" as reminder that we concatenated multiple datasets
+            ms_name = '{}{}.ms'.format(dataset_basename,
+                                       "" if len(args) == 1 else ".et_al")
```

Comment:
This filename generation logic looks like it could be in a function. The script itself has become quite large so it might be nice to start the process of moving this kind of stuff into individual modules.

Reply:
The relevant individual module is called `ms_extra` and it already has more lines than the script 😛 

There is still a lot of dead code in both `ms_extra` and `mvftoms` (like the cal table generation), so I'm not too worried about the line count. I'll still move it to a function though - that sounds good.